plugins/lua/lua_plugin.c

```diff
@@ -983,7 +983,7 @@ static int uwsgi_api_register_signal(lua_State *L) {
 		uwsgi_lock(uwsgi.signal_table_lock);
 
 		// master-proc checks receiver
-		strcpy(use->receiver, who);
+		strncpy(use->receiver, who, len);
```

Comment:
Il 07/12/2015 19:50, Darvame Hleran ha scritto:
> In plugins/lua/lua_plugin.c
> <https://github.com/unbit/uwsgi/pull/1084#discussion_r46859422>:
>
>> @@ -983,7 +983,7 @@ static int uwsgi_api_register_signal(lua_State *L) {
>>  		uwsgi_lock(uwsgi.signal_table_lock);
>>
>>  		// master-proc checks receiver
>> -		strcpy(use->receiver, who);
>> +		strncpy(use->receiver, who, len);
>
> Sorry, for slow reaction, but
>
> strncpy(use->receiver, who, len);
>
> lua provides len without terminator, so we need do
>
> strncpy(use->receiver, who, len +1);//also copy terminator from 'who'
>
> or
>
> |strcpy(use->receiver, who); //len check has been passed, 'who' has terminator |

Thanks fixed. Forgot to give you proper credit, sorry.


-- 
Riccardo Magliocchetti
@rmistaken

http://menodizero.it